pywinauto/linux/application.py

```diff
@@ -120,13 +121,33 @@ def cpu_usage(self, interval=None):
         if not self.process:
             raise AppNotConnected("Please use start or connect before trying "
                                   "anything else")
-        if interval:
-            time.sleep(interval)
+        proc_pid_stat = "/proc/{}/stat".format(self.process)
+
+        def read_cpu_info():
+            with open(proc_pid_stat, 'r') as s:
+                pid_info = s.read().split()
+            with open("/proc/stat") as s:
+                info = s.read().split()
+            # return a tuple as following:
+            # pid utime, pid stime, total utime, total stime
+            return (int(pid_info[13]), int(pid_info[14]), int(info[1]), int(info[3]))
+
         try:
-            proc_info = subprocess.check_output(["ps", "-p", str(self.process), "-o", "%cpu"], universal_newlines=True)
-            proc_info = proc_info.split("\n")
-            return float(proc_info[1])
-        except Exception:
+            before = read_cpu_info()
+            if not interval:
+                interval = Timings.cpu_usage_interval
+            time.sleep(interval)
+            after = read_cpu_info()
+            pid_time = (after[0] - before[0]) + (after[1] - before[1])
+            sys_time = (after[2] - before[2]) + (after[3] - before[3])
+            if not sys_time:
+                res = 0.0
+            else:
+                res = 100.0 * (float(pid_time) / float(sys_time))
```

Comment:
Maybe I misunderstand something... In my mind `sys_time` is how much time the app spent in the Linux kernel (in dependent kernel modules) and `pid_time` is how much time the app spent in user space modules. So the formula should be a sum of `sys_time` and `pid_time` divided by `interval`. Nope?

Reply:
Yes, you got confused with the variable name. Pls, have a look at `read_cpu_info` helper. It returns four values: user time and kernel times for pid and for all system. So the `pid_time` keeps user and kernel time of the process, while `sys_time` stores user time and kernel time of the system. I should rename it to `total_time`

Reply:
OK, now got it.